app/controllers/posts_controller.rb

```diff
@@ -5,6 +5,26 @@ def index
     @posts = @user.posts
   end
 
+  def create
+    user = ApplicationController.current_user
+    input = params.require(:new_post).permit(:title, :text)
+    new_post =
+      Post.new(
+        author: user,
+        title: input['title'],
+        text: input['text'],
+        likes_counter: 0,
+        comments_counter: 0
+      )
+
+    if new_post.save
+      flash[:success] = 'posted successfully'
+      redirect_to user_post_path(user.id, new_post.id)
+    else
+      flash[:alert] = 'error creating post'
+    end
+  end
+
```

Comment:
@CollinsTatang  Hi, i need some clarification on creating a `new` action for posts. I am using the new post form as a partial in `users/index` page and not serving the form in a new html page. wouldn't `create` action be sufficient for this.

Reply:
Hi @ram1117 
It could be sufficient, but according to the project requirement you are expected to create a separate controller action for create and new method.